stream_alert_cli/terraform_generate.py

```diff
@@ -352,6 +352,32 @@ def generate_cloudtrail(cluster_name, cluster_dict, config):
     """
     modules = config['clusters'][cluster_name]['modules']
     cloudtrail_enabled = bool(modules['cloudtrail']['enabled'])
+    existing_trail_default = False
+    existing_trail = modules['cloudtrail'].get('existing_trail', existing_trail_default)
+    is_global_trail_default = True
+    is_global_trail = modules['cloudtrail'].get('is_global_trail', is_global_trail_default)
+    event_pattern_default = {
+        'account': [config['global']['account']['aws_account_id']]
+    }
+    event_pattern = modules['cloudtrail'].get('event_pattern', event_pattern_default)
+
+    # From here:
+    # http://docs.aws.amazon.com/AmazonCloudWatch/latest/events/CloudWatchEventsandEventPatterns.html
+    valid_event_pattern_keys = {
+        'version',
+        'id',
+        'detail-type',
+        'source',
+        'account',
+        'time',
+        'region',
+        'resources',
+        'detail'
+    }
+    if not set(event_pattern.keys()).issubset(valid_event_pattern_keys):
+        LOGGER_CLI.error('Invalid CloudWatch Event Pattern!')
+        sys.exit(1)
```

Comment:
We should probably start returning a value in places like this (maybe just a simple bool == False, or a more comprehensive solution would be returning a non-zero error number) that we could check up the call stack and exit on error. It would avoid having `sys.exit(1)` calls scattered about the CLI and could allow for us to start using useful exit codes for certain errors (that are not `1`). Not a necessary change now, but something to keep in mind.